src/olympia/amo/management/commands/initialize_data.py

```diff
@@ -0,0 +1,201 @@
+                try:
+                    return func(*args, **kwargs)
+                except Exception as e:
+                    if x == retries:
+                        raise
+                    else:
+                        sleep = backoff_in_seconds * 2**x + random.uniform(0, 1)
+                        logging.warning(
+                            f'{wrapper.__name__} failed. '
+                            f'Retrying in {sleep:.2f} seconds... Error: {str(e)}'
+                        )
+                        time.sleep(sleep)
+                        x += 1
+
+        return wrapper
+
+    return decorator
+
+
+class Command(BaseCommand):
+    """
+    Creates the database for this project.
+
+    This command is idempotent and will not re-create the database if it already exists.
+    It will also not re-seed the database or reindex the data
+    if the database already exists.
+    """
+
+    help = 'Creates, seeds, and indexes the database for this project.'
+    connection = None
+    db_info = None
+    db_exists = False
+    num_addons = 10
+    num_themes = num_addons
+
+    def __init__(self, *args, **options):
+        super().__init__(*args, **options)
+
+        self.db_info = settings.DATABASES.get('default')
+        self.connection = self.connect_to_db()
+        self.db_exists = self.check_db_exists()
+
+    def add_arguments(self, parser):
+        super().add_arguments(parser)
+        parser.add_argument(
+            '--force-db', action='store_true', help='Force creating the database'
+        )
+        parser.add_argument(
+            '--skip-seed',
+            action='store_true',
+            help='Skip seeding the database with addons',
+        )
+        parser.add_argument(
+            '--skip-index', action='store_true', help='Skip indexing the database'
+        )
+
+    @retry_with_backoff(retries=3, backoff_in_seconds=1)
+    def connect_to_db(self):
+        engine = self.db_info.get('ENGINE').split('.')[-1]
+        if engine != 'mysql':
+            raise CommandError('create_db only supports mysql databases')
+
+        kwargs = {
+            'user': self.db_info.get('USER'),
+            'passwd': self.db_info.get('PASSWORD'),
+            'host': self.db_info.get('HOST'),
+        }
+        if self.db_info.get('PORT'):
+            kwargs['port'] = int(self.db_info.get('PORT'))
+
+        logging.info('connecting to db')
+        return mysql.connect(**kwargs)
+
+    @retry_with_backoff(retries=3, backoff_in_seconds=1)
+    def check_db_exists(self):
+        try:
+            self.connection.select_db(self.db_info.get('NAME'))
+            return True
+        except mysql.Error as exc:
+            logging.info(exc)
+            return False
+
+    def create_db(self):
+        logging.info('Cleaning up directories linked to database records...')
+        root = os.path.join('/', 'data', 'olympia')
+        clean_dirs = (
+            os.path.join(root, 'user-media'),
```

Comment:
I think it's a good point that we should synchronize the storage directory more with the database. 

That being said, those issues existed before. Makeup will not nuke anything unless you ask it to.

Can you think of a way to improve this or what should we do with this comment?

Reply:
I guess `make data_export` should copy the files too?  I don't know what Mat thinks but I lean towards this being a follow-up issue.

Reply:
100% follow up. Yeah I think the "proper" solution is to merge the "storage" and "data_mysql" volumes but this would need to be looked at. copying the storage is also an idea.

Reply:
(I suspect using Django's native `dumpdata` is going to be tricky - we have some complex models & relationships - Django might need a lot of hints to dump/load that data, and it might also be a lot slower)